Mathlib/Data/ENNReal/Inv.lean

```diff
@@ -91,11 +91,48 @@ protected theorem mul_inv_cancel (h0 : a ≠ 0) (ht : a ≠ ∞) : a * a⁻¹ =
 protected theorem inv_mul_cancel (h0 : a ≠ 0) (ht : a ≠ ∞) : a⁻¹ * a = 1 :=
   mul_comm a a⁻¹ ▸ ENNReal.mul_inv_cancel h0 ht
 
-protected theorem div_mul_cancel (h0 : a ≠ 0) (hI : a ≠ ∞) : b / a * a = b := by
-  rw [div_eq_mul_inv, mul_assoc, ENNReal.inv_mul_cancel h0 hI, mul_one]
+protected lemma inv_mul_cancel_left (ha₀ : a = 0 → b = 0) (ha : a = ∞ → b = 0) :
+    a⁻¹ * (a * b) = b := by
+  obtain rfl | ha₀ := eq_or_ne a 0
+  · simp_all
+  obtain rfl | ha := eq_or_ne a ⊤
+  · simp_all
+  · simp [← mul_assoc, ENNReal.inv_mul_cancel, *]
 
-protected theorem mul_div_cancel' (h0 : a ≠ 0) (hI : a ≠ ∞) : a * (b / a) = b := by
-  rw [mul_comm, ENNReal.div_mul_cancel h0 hI]
+protected lemma mul_inv_cancel_left (ha₀ : a = 0 → b = 0) (ha : a = ∞ → b = 0) :
+    a * (a⁻¹ * b) = b := by
```

Comment:
We discussed this offline, but don't forget to add the version of this where you assume a != 0 and a != top